utils/json-utils/src/main/scala/pl/touk/nussknacker/engine/json/swagger/extractor/LazyJsonTypedMap.scala

```diff
@@ -0,0 +1,117 @@
+package pl.touk.nussknacker.engine.json.swagger.extractor
+
+import com.github.benmanes.caffeine.cache.{CacheLoader, Caffeine, LoadingCache}
+import io.circe.syntax.EncoderOps
+import io.circe.{Json, JsonObject}
+import pl.touk.nussknacker.engine.json.swagger._
+
+import pl.touk.nussknacker.engine.json.swagger.extractor.JsonToNuStruct.JsonToObjectError
+
+import java.{util => ju}
+import java.util.stream.Collectors.toList
+import scala.util.Try
+import cats.implicits._
+
+import java.util.concurrent.CompletionException
+
+class LazyJsonTypedMap(jsonObject: JsonObject, definition: SwaggerObject, path: String = "")
+    extends java.util.Map[String, Any] {
+
+  def this() = this(JsonObject.empty, SwaggerObject(Map.empty, AdditionalPropertiesDisabled, List.empty))
+
+  import LazyJsonTypedMap._
+  import scala.jdk.CollectionConverters._
+
+  private val loader: CacheLoader[String, Option[Any]] = key => Some(extractValue(key))
```

Comment:
Why not use here normal in memory map instead of Caffeine?

Reply:
Caffeine is in the memory. It is lazy map - how would you handle this laziness without caffeine?

Reply:
Ahh, we could use a mutable map.. Let's leave it.